File: Problems 3200-3299/3254. Find the Power of K-Size Subarrays I/solution.py
class Solution:
    def resultsArray(self, nums: List[int], k: int) -> List[int]:
        res = []
        numConsec = 0
        for i, num in enumerate(nums):
            if i == 0 or nums[i-1] == nums[i] - 1:
                numConsec += 1
            else:
                numConsec = 1
            
            if i >= k-1:
                if numConsec >= k:
                    res.append(nums[i])
                else:
                    res.append(-1)
        return res

            

File: Problems 3200-3299/3254. Find the Power of K-Size Subarrays I/solution.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) {
        int count = 0;
        vector<int> res;
        for (int i = 0; i < nums.size(); i++) {
            if (i == 0 || (i > 0 && nums[i] == nums[i-1]+1)) {
                count++;
            } else {
                count = 1;
            }

            if (i >= k-1) {
                if (count >= k) {
                    res.push_back(nums[i]);
                } else {
                    res.push_back(-1);
                }
            }
        }
        return res;
    }
};